library/system/test/commented_config_file_test.rb

```diff
@@ -0,0 +1,314 @@
+      subject.line_comment = "line comment"
+      subject.comment_before = "# comment\n# lines \n# before\n"
+      expect(subject.format).to eq line
+      expect(subject.to_s).to eq line
+    end
+  end
+
+  context "Low-level parser" do
+    describe "#comment_line?" do
+      subject { described_class.new }
+
+      context "with the default '#' comment marker" do
+        it "Detects a simple comment line" do
+          expect(subject.comment_line?("# foo")).to eq true
+          expect(subject.comment_line?("#foo")).to eq true
+        end
+
+        it "Can handle leading whitespace" do
+          expect(subject.comment_line?("  # foo")).to eq true
+        end
+
+        it "Can handle trailing whitespace" do
+          expect(subject.comment_line?("# foo  \n")).to eq true
+        end
+
+        it "Rejects non-comment lines" do
+          expect(subject.comment_line?("foo")).to eq false
+          expect(subject.comment_line?("  foo")).to eq false
+          expect(subject.comment_line?("foo # bar")).to eq false
+          expect(subject.comment_line?("// foo")).to eq false
+          expect(subject.comment_line?("")).to eq false
+        end
+      end
+
+      context "with a custom '//' comment marker" do
+        subject do
+          ccf = described_class.new
+          ccf.comment_marker = "//"
+          ccf
+        end
+
+        it "Detects a simple comment line" do
+          expect(subject.comment_line?("// foo")).to eq true
+          expect(subject.comment_line?("//foo")).to eq true
+        end
+
+        it "Can handle leading whitespace" do
+          expect(subject.comment_line?("  // foo")).to eq true
+        end
+
+        it "Can handle trailing whitespace" do
+          expect(subject.comment_line?("// foo  \n")).to eq true
+        end
+
+        it "Rejects non-comment lines" do
+          expect(subject.comment_line?("foo")).to eq false
+          expect(subject.comment_line?("  foo")).to eq false
+          expect(subject.comment_line?("foo # bar")).to eq false
+          expect(subject.comment_line?("# foo")).to eq false
+          expect(subject.comment_line?("")).to eq false
+        end
+      end
+    end
+
+    describe "#empty_line?" do
+      subject { described_class.new }
+
+      it "Detects a completely empty line" do
+        expect(subject.empty_line?("")).to eq true
+      end
+
+      it "Detects lines with only whitespace " do
+        expect(subject.empty_line?(" ")).to eq true
+        expect(subject.empty_line?("  ")).to eq true
+        expect(subject.empty_line?(" \n")).to eq true
+        expect(subject.empty_line?("\t")).to eq true
+        expect(subject.empty_line?("\t\n  \t\n")).to eq true
+      end
+
+      it "Rejects non-empty lines" do
+        expect(subject.empty_line?("x")).to eq false
+        expect(subject.empty_line?("  x")).to eq false
+        expect(subject.empty_line?("  x  ")).to eq false
+        expect(subject.empty_line?("  \nx  ")).to eq false
```

Comment:
`"\nx"` or `"x\n"` might be useful to test as well...

Reply:
Done.